core/node/libp2p/pubsub.go

```diff
@@ -1,26 +1,75 @@
 package libp2p
 
 import (
+	"context"
+	"errors"
+	"fmt"
+
+	datastore "github.com/ipfs/go-datastore"
 	pubsub "github.com/libp2p/go-libp2p-pubsub"
 	"github.com/libp2p/go-libp2p/core/discovery"
 	"github.com/libp2p/go-libp2p/core/host"
+	"github.com/libp2p/go-libp2p/core/peer"
 	"go.uber.org/fx"
 
 	"github.com/ipfs/kubo/core/node/helpers"
+	"github.com/ipfs/kubo/repo"
 )
 
+type P2PPubSubIn struct {
+	fx.In
+
+	Repo      repo.Repo
+	Host      host.Host
+	Discovery discovery.Discovery
+}
```

Comment:
What is the point of `P2PPubSubIn` ? Couldn't you use arguments on the anonymous functions ?

Reply:
Its 3 of them, this is more tidy and it costs nothing.